core/src/banking_stage.rs

```diff
@@ -452,9 +452,11 @@ impl BankingStage {
 
                 let mut packet_receiver = PacketReceiver::new(id, packet_receiver);
                 let poh_recorder = poh_recorder.clone();
-                let transaction_status_sender = transaction_status_sender.clone();
-                let replay_vote_sender = replay_vote_sender.clone();
 
+                let committer = Committer::new(
+                    transaction_status_sender.clone(),
+                    replay_vote_sender.clone(),
+                );
```

Comment:
nit - instead of passing 2 senders around, one `committer` is passed around. So code-wise, 50% improvement. But  in prev code, it is obvious that senders are passed in from `BankingStage::new()` so they were just being passed around till final destination. In new version, fresh reader probably would wonder what is `committer` and why it is passed around. This is minor inconvenience, just if the senders were bundled before being passed into BankingStage, it might be cleaner.

Reply:
Thanks for explaining, it makes more sense with context 😄 

For sake of discussing interfaces, since the `sender`s are only used by `committer`, one can argue they don't need to be visible to `banking_stage` at all. If uses factory pattern, they would be built into `committer`, which in turn be passed into `banking_stage`. But then you have to touch caller of `banking_stage`, so what you did here is just fine, unless you want to make `committer` a member of `banking_stage` ...

Reply:
> @ryoqun - I know you love discussing interfaces! Feel free to chime in here :smiley: 

@apfitzge i 100%-agree with that opinion. thanks for putting the context into words. yeah, the goal of pluggability of `BankingStage` is important to keep it in mind for this whole of refactoring.

not said above, but there's also an added bonus of unit-test friendliness, thanks to this pr, although `Committer`'s state is small (just 2 fields).

Overall, i think these abstractions could be justified.